tests/webtests/EntityEditTest.php

```diff
@@ -81,6 +82,33 @@ public function test_it_renders_the_form()
         );
     }
 
+    public function test_it_hides_name_id_format_unspecified()
+    {
+        // 1. When NameIdFormat is not Unspecified, the form field does appear on the form
+        $crawler = self::$pantherClient->request('GET', "/entity/edit/test/{$this->manageId}/1");
+        $this->assertOnPage(
+            'Name id format',
+            $crawler
+        );
+        // 2. But when the nameIdFormat is unspecified, the form field is not displayed on the form
+        $this->registerManageEntity(
+            'test',
+            'saml20_sp',
+            '88888888-1111-1111-1111-888888888888',
+            'SP1',
+            'https://spx-entityid.example.com',
+            'https://spx-entityid.example.com/metadata',
+            WebTestFixtures::TEAMNAME_SURF,
+            '12',
+            Constants::NAME_ID_FORMAT_UNSPECIFIED
+        );
+        $crawler = self::$pantherClient->request('GET', "/entity/edit/test/88888888-1111-1111-1111-888888888888/1");
```

Comment:
This does not have to be with a full blown panther instance I guess, could also be done with lightweigth kernelbrowser